easybuild/easyblocks/s/superlu.py

```diff
@@ -0,0 +1,148 @@
+        extra_vars = {
+            'build_shared_libs': [False, "Build shared library (instead of static library)", CUSTOM],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """
+        Set the CMake options for SuperLU
+        """
+        self.cfg['separate_build_dir'] = True
+
+        if self.cfg['build_shared_libs']:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=ON')
+        else:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=OFF')
+
+        # Add -fPIC flag if necessary
+        if self.toolchain.options['pic']:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=ON')
+        else:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=OFF')
+
+        # Make sure not to build the slow BLAS library included in the package
+        self.cfg.update('configopts', '-Denable_blaslib=OFF')
+
+        # Set the BLAS library to use
+        # For this, use the BLA_VENDOR option from the FindBLAS module of CMake
+        # Check for all possible values at https://cmake.org/cmake/help/latest/module/FindBLAS.html
+        toolchain_blas = self.toolchain.definition().get('BLAS', ['NO_BLAS_IN_TOOLCHAIN'])[0]
+        if toolchain_blas == 'imkl':
+            imkl_version = get_software_version('imkl')
+            if LooseVersion(imkl_version) >= LooseVersion('10'):
+                # 'Intel10_64lp' -> For Intel mkl v10 64 bit,lp thread model, lp64 model
+                # It should work for Intel MKL 10 and above, as long as the library names stay the same
+                # SuperLU requires thread, 'Intel10_64lp_seq' will not work!
+                self.cfg.update('configopts', '-DBLA_VENDOR="Intel10_64lp"')
+            else:
+                # 'Intel' -> For older versions of mkl 32 and 64 bit
+                self.cfg.update('configopts', '-DBLA_VENDOR="Intel"')
+        elif toolchain_blas == 'ACML':
+            self.cfg.update('configopts', '-DBLA_VENDOR="ACML"')
+        elif toolchain_blas == 'ATLAS':
+            self.cfg.update('configopts', '-DBLA_VENDOR="ATLAS"')
+        elif toolchain_blas == 'OpenBLAS':
+            # Unfortunately, OpenBLAS is not recognized by FindBLAS from CMake,
+            # we have to specify the OpenBLAS library manually
+            self.cfg.update('configopts', '-DBLAS_LIBRARIES="${EBROOTOPENBLAS}/lib/libopenblas.a;-pthread"')
+        elif toolchain_blas == 'NO_BLAS_IN_TOOLCHAIN':
+            # This toolchain has no BLAS library, let CMake find one
+            self.cfg.update('configopts', '-DBLA_VENDOR="All"')
+        else:
+            # This BLAS library is not supported yet
+            raise EasyBuildError("BLAS library '%s' is not supported yet" % toolchain_blas, err)
+
+        super(EB_SuperLU, self).configure_step()
+
+    def test_step(self):
+        """
+        Run the testsuite of SuperLU
+        """
+        self.cfg['runtest'] = "test"
```

Comment:
this hard forces always running the tests, maybe you only want to set this if `runtest` was not specified in the easyconfig file (so you can disable running the tests from there):

```python
if self.cfg['runtest'] is None:
     self.cfg['runtest'] = 'test'
```